app/controllers/studentLoans/PglAmountController.scala

```diff
@@ -0,0 +1,105 @@
+import forms.{AmountForm, FormUtils}
+import models.User
+import controllers.studentLoans.routes.StudentLoansCYAController
+import models.mongo.EmploymentUserData
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import repositories.EmploymentUserDataRepository
+import services.EmploymentSessionService
+import services.studentLoans.StudentLoansService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.{Clock, SessionHelper}
+import views.html.studentLoans.PglAmountView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class PglAmountController @Inject()(implicit val mcc: MessagesControllerComponents,
+                                    authAction: AuthorisedAction,
+                                    val employmentSessionService: EmploymentSessionService,
+                                    val studentLoansService: StudentLoansService,
+                                    employmentUserDataRepository: EmploymentUserDataRepository,
+                                    view: PglAmountView,
+                                    appConfig: AppConfig,
+                                    errorHandler: ErrorHandler,
+                                    ec: ExecutionContext,
+                                    clock: Clock) extends FrontendController(mcc) with I18nSupport with SessionHelper with FormUtils {
+
+  def show(taxYear: Int, employmentId: String): Action[AnyContent] = (authAction andThen TaxYearAction.taxYearAction(taxYear)).async { implicit user =>
+
+    if (appConfig.studentLoansEnabled) {
+      employmentUserDataRepository.find(taxYear, employmentId).map {
+        case Left(_) => errorHandler.internalServerError()
+        case Right(optionCyaData) =>
+          optionCyaData match {
+            case Some(cyaData) =>
+              val pglAmount = cyaData.employment.studentLoans.flatMap(_.pglDeductionAmount)
+              val employerName = cyaData.employment.employmentDetails.employerName
+
+              val form = pglAmount.fold(amountForm(employerName)
+              )(amountForm(employerName).fill _)
+              Ok(view(taxYear, form, employmentId, employerName))
+            case None => Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear))
+          }
+
+      }
+    }
+
+    else {
+      Future.successful(Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear)))
+    }
+  }
+
+  def submit(taxYear: Int, employmentId: String): Action[AnyContent] = (authAction andThen TaxYearAction.taxYearAction(taxYear)).async { implicit user =>
+
+    if (appConfig.studentLoansEnabled) {
+      val redirectUrl = StudentLoansCYAController.show(taxYear, employmentId).url
+      employmentSessionService.getSessionDataAndReturnResult(taxYear, employmentId)(redirectUrl) { cya =>
+        amountForm(cya.employment.employmentDetails.employerName).bindFromRequest().fold(
```

Comment:
Same as above but for submit method. Think we should check that the user can said yes to pglDeduction before allowing them to add an amount for it. If they shouldnt be here can redirect to the student loans cya page I'd say.